vignettes/speedup_options.Rmd

```diff
@@ -0,0 +1,458 @@
+  comment = "#>",
+  fig.width = 7.5,
+  fig.height = 7.5,
+  fig.path = "vignettes/speedup-options-"
+)
+set.seed(9876)
+```
+
+```{r packages}
+library(EpiNow2)
+library(scoringutils)
+library(data.table)
+library(rstan)
+library(cmdstanr)
+library(ggplot2)
+library(lubridate)
+library(scales)
+library(posterior)
+```
+
+This vignette explores available model options in _EpiNow2_ and how they vary by speed and accuracy of estimates. We will compare the models by run time and accuracy of estimates (quantitatively and qualitatively) against the "true" trajectory of the data.
+
+## The benchmarking data
+
+To compare the model options, we will need a dataset for which we know the "true" values and trajectories. So, we will start by generating the "true" infections and $R_t$ data using _EpiNow2_'s `forecast_infections()` function. 
+
+`forecast_infections()` requires a fitted estimates object from `epinow()` with `output` set to "fit", the trajectory of the reproduction number, `R`, and the number of samples to simulate. So, we will set these up first.
+
+To obtain the `estimates` object, we will run the `epinow()` function using real-world observed data and delay distributions to recover realistic parameter values. We will use the first $100$ observations of the `example_confirmed` data set, the `example_generation_time`, `example_incubation_period`, and `example_reporting_delay` values that come with _EpiNow2_.
+
+Several of the argument values we will use here will be kept the same for all the model runs, so we will set them up here. These include the generation time, incubation period, reporting delay, observation model options, and the forecast horizon.
+```{r, shared_inputs}
+# Set the number of cores to use
+options(mc.cores = 4)
+
+# Generation time
+generation_time <- Gamma(
+  shape = Normal(1.3, 0.3),
+  rate = Normal(0.37, 0.09),
+  max = 14
+)
+
+# Incubation period
+incubation_period <- LogNormal(
+  meanlog = Normal(1.6, 0.05),
+  sdlog = Normal(0.5, 0.05),
+  max = 14
+)
+
+# Reporting delay
+reporting_delay <- LogNormal(
+  meanlog = 0.5,
+  sdlog = 0.5,
+  max = 10
+)
+
+# Combine the incubation period and reporting delay into one delay
+delay <- incubation_period + reporting_delay
+
+# Observation model options
+obs <- obs_opts(
+  scale = list(mean = 0.1, sd = 0.025),
+  return_likelihood = TRUE
+)
+
+# No forecasting; just estimation
+horizon <- 0
+```
+
+Now, let's generate the `estimates` object from `epinow()`.
+```{r estimates}
+estimates <- epinow(
+  data = example_confirmed[1:100],
+  generation_time = generation_time_opts(example_generation_time),
+  delays = delay_opts(example_incubation_period + example_reporting_delay),
+  rt = rt_opts(prior = list(mean = 2, sd = 0.1), rw = 14),
+  gp = NULL,
+  obs = obs,
+  horizon = horizon,
+  output = "fit"
+)
+```
+
+For the `R` data, we will set up an arbitrary trajectory and add some Gaussian noise.
+```{r R-data}
+# Arbitrary reproduction number trajectory
+R <- c(
```

Comment:
as this test R is non-stationary and drawn from model different data generating processes (i.e stepwise changes, linear changes etc) I think it somewhat biases you towards less mechanistic model options (more because there is less room for mechanistic choices to do well as none of the latent process models are well defined to fit to this data). Probably should just note this complexity.

Reply:
Thanks. I've added a caveat here bc56702e03fe2bf850ed30966e731bde94bb416e.